src/com/company/A/S1903215.java

```diff
@@ -102,6 +102,7 @@ private void insertDNASeries(String DNASeries) {
           if (curPointer.DNASeries != null) {
             appearingDNASeries.add(curPointer.DNASeries);
           }
+          pointers.set(i, curPointer);
```

Comment:
Not necessary since we are changing the object?

Reply:
Similar issue,  curpointer point to the pointer in the array , and then curpointer points to thepointer's children, the pointer itself in the array stays the same. It is not changed. 

Also, I've created an iterator in the files after refactoring, you can checkout for better test.